GVFS/GVFS.Common/FileSystem/IVolumeStateWatcher.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System;
+
+namespace GVFS.Common.FileSystem
+{
+    /// <summary>
+    /// Monitors the system for changes to the state of any disk volume, notifying subscribers when a change occurs.
+    /// </summary>
+    public interface IVolumeStateWatcher : IDisposable

```

Comment:
Today's behavior is: one and only one best effort attempt to automount after initial login. I don't think we should expand the scope of that feature in _this_ PR, but should instead focus on extending that same behavior to BitLockered drives. 

Expanding the intent of the feature could also be a fine thing to do, but it carries a lot more risk (we have nothing today that will automount without either an explicit user gesture or a fatal error, and nothing that will automount at any time other than initial login) and so it could lead to surprising interactions for users and issues that are difficult to diagnose. So going after those other scenarios could be nice, but will require a much more thorough design to ensure we don't create any unexpected behaviors.